### PR TITLE
feat: add karate e2e test on crud operations for one book

### PR DESCRIPTION
**Pull request by Annushkaaaaa**

### CODE REVIEW

File: Api/Properties/launchSettings.json [hunk 1]
```diff
@@ -5,7 +5,7 @@
       "launchBrowser": true,
       "launchUrl": "swagger",
       "environmentVariables": {
-        "ASPNETCORE_ENVIRONMENT": "MockForDevelopment"
+        "ASPNETCORE_ENVIRONMENT": "MockForPullRequest"
```
Comment (fpandyz) [inline Api/Properties/launchSettings.json:8]: Why did you change it?

File: docker-compose.yml [hunk 2]
```diff
@@ -16,19 +16,51 @@ services:
...
+    depends_on:
+      - books-db
+    profiles: ['MockForPullRequest']
+    extends:
+      service: books-api
+    environment:
+      - ASPNETCORE_ENVIRONMENT=MockForPullRequest
+      - ASPNETCORE_HTTP_PORTS=80
+  
+  books-api-for-develop:
```
Comment (fpandyz) [inline docker-compose.yml:43]: Why did you rename it?

File: docker-compose.yml [hunk 3]
```diff
@@ -16,19 +16,51 @@ services:
...
+    container_name: 'compensations-api-for-e2e'
```
Comment (fpandyz) [inline docker-compose.yml:44]: This is a wrong container name: copy/paste mistake.

File: README.md [hunk 4]
```diff
@@ -16,6 +16,29 @@ After this command, start the application from IDE.
...
+docker-compose --profile MockForPullRequest up -d
+```
+
+Go to VS Code
+1. Install the extension "Dev Containers" (extension id: ms-vscode-remote.remote-containers)
+2. Click on the blue button in the lower left corner of your screen
+3. Click "Rebuild Container" or something like this - the project will start in dev container
+4. Enter this command to run the tests
+```
+java -jar /karate.jar .
```
Comment (fpandyz) [inline README.md:33]: If I follow instructions tests fail locally with the following exception:
`root@docker-desktop:/workspaces/inner-circle-books-api# java -jar /karate.jar .
04:47:41.800 [main]  INFO  com.intuit.karate - Karate version: 1.4.1
04:47:42.901 [main]  DEBUG com.intuit.karate - request:
1 > POST http://localhost:1080/auth/login
1 > Content-Type: application/json; charset=UTF-8
1 > Content-Length: 30
1 > Host: localhost:1080
1 > Connection: Keep-Alive
1 > User-Agent: Apache-HttpClient/4.5.14 (Java/11.0.16)
1 > Accept-Encoding: gzip,deflate
{"login":null,"password":null}

04:47:42.913 [main]  ERROR com.intuit.karate - org.apache.http.conn.HttpHostConnectException: Connect to localhost:1080 [localhost/127.0.0.1, localhost/0:0:0:0:0:0:0:1] failed: Connection refused (Connection refused), http call failed after 152 milliseconds for url: http://localhost:1080/auth/login
04:47:42.923 [main]  ERROR com.intuit.karate - E2E/CRUD operations E2E.feature:21
And method POST
http call failed after 152 milliseconds for url: http://localhost:1080/auth/login
org.apache.http.conn.HttpHostConnectException: Connect to localhost:1080 [localhost/127.0.0.1, localhost/0:0:0:0:0:0:0:1] failed: Connection refused (Connection refused)
E2E/CRUD operations E2E.feature:21
---------------------------------------------------------
feature: E2E/CRUD operations E2E.feature
scenarios:  1 | passed:  0 | failed:  1 | time: 0.3144
---------------------------------------------------------

04:47:43.446 [main]  INFO  com.intuit.karate.Suite - <<fail>> feature 1 of 1 (0 remaining) E2E/CRUD operations E2E.feature
Karate version: 1.4.1
======================================================
elapsed:   1.56 | threads:    1 | thread time: 0.31 
features:     1 | skipped:    0 | efficiency: 0.20
scenarios:    1 | passed:     0 | failed: 1
======================================================
>>> failed features:
http call failed after 152 milliseconds for url: http://localhost:1080/auth/login
org.apache.http.conn.HttpHostConnectException: Connect to localhost:1080 [localhost/127.0.0.1, localhost/0:0:0:0:0:0:0:1] failed: Connection refused (Connection refused)
E2E/CRUD operations E2E.feature:21
<<<

HTML report: (paste into browser to view) | Karate version: 1.4.1
file:///workspaces/inner-circle-books-api/target/karate-reports/karate-summary.html
===================================================================

com.intuit.karate.KarateException: there are test failures !
        at ...(.)`

here are my logs from docker-compose
`PS C:\Projects\Tourmaline\inner-circle-books-api> docker-compose --profile MockForPullRequest up
[+] Running 8/8
 ✔ mockServer Pulled                                                                                              18.3s
   ✔ fc251a6e7981 Pull complete                                                                                    2.2s
   ✔ fda4ba87f6fb Pull complete                                                                                    2.4s
   ✔ a1f1879bb7de Pull complete                                                                                    2.5s
   ✔ 407f249037ea Pull complete                                                                                    3.9s
   ✔ 8a4811140f89 Pull complete                                                                                   14.8s
   ✔ 3b9f57d87d13 Pull complete                                                                                   15.0s
   ✔ 03ef7d409ab5 Pull complete                                                                                   15.1s
[+] Building 124.7s (19/19) FINISHED                                                               docker:desktop-linux
 => [books-api-for-e2e internal] load build definition from Dockerfile                                             0.0s
 => => transferring dockerfile: 526B                                                                               0.0s
 => [books-api-for-e2e internal] load metadata for mcr.microsoft.com/dotnet/sdk:8.0                                2.1s
 => [books-api-for-e2e internal] load metadata for mcr.microsoft.com/dotnet/aspnet:8.0                             2.2s
 => [books-api-for-e2e internal] load .dockerignore                                                                0.0s
 => => transferring context: 358B                                                                                  0.0s
 => [books-api-for-e2e build 1/7] FROM mcr.microsoft.com/dotnet/sdk:8.0@sha256:032381bcea86fa0a408af5df63a930f1f  77.1s
 => => resolve mcr.microsoft.com/dotnet/sdk:8.0@sha256:032381bcea86fa0a408af5df63a930f1ff5b03116c940a7cd744d3b648  0.0s
 => => sha256:032381bcea86fa0a408af5df63a930f1ff5b03116c940a7cd744d3b648e66749 1.08kB / 1.08kB                     0.0s
 => => sha256:bb3f2b52e6af242cee1bc6c19ce79e05544f8a1d13f5a6c1e828d98d2dbdc94e 28.06MB / 28.06MB                  21.3s
 => => sha256:11e53e0176cdd626c2032644dddb5a96e6eaf1b0fffe690b7158ce941c258190 3.33kB / 3.33kB                     0.7s
 => => sha256:52e5289568e1a9c465e5a6929dc8de218cd370ffe78b10096254d2dc0b382293 2.22kB / 2.22kB                     0.0s
 => => sha256:42e260472f7d10de86d67d6006c5cc67e055e765d61a13b327ee642c4e1a8835 5.58kB / 5.58kB                     0.0s
 => => sha256:575ae10faf5887a49d34f7131754099bcd5714cb755499cb0a0d3a4de8c9dd2c 18.50MB / 18.50MB                  15.1s
 => => sha256:a99f979d704a240405f08913e30302ea667012268e72f3c676dc942e1665f7e4 30.82MB / 30.82MB                  26.9s
 => => sha256:963ffc7a68ff07ad0a48fc7f90d20c556c444bbabfb4bd45a180cdcba6510210 164B / 164B                        15.3s
 => => sha256:4d23af0f8da9b6b2cf8447d9208e571426772a573abe3537b5553f9835984251 10.74MB / 10.74MB                  17.5s
 => => sha256:8a00ebd09a564c5b434947515b324fe3154f5fd84ef45409580a364ce1accbf1 31.29MB / 31.29MB                  27.2s
 => => sha256:f8f1d29d08263d7836d1ac8dcc64734c2198079928f87dc7e35f06b68e9e0e40 170.85MB / 170.85MB                72.1s
 => => extracting sha256:bb3f2b52e6af242cee1bc6c19ce79e05544f8a1d13f5a6c1e828d98d2dbdc94e                          1.7s
 => => extracting sha256:575ae10faf5887a49d34f7131754099bcd5714cb755499cb0a0d3a4de8c9dd2c                         99.3s
 => => extracting sha256:11e53e0176cdd626c2032644dddb5a96e6eaf1b0fffe690b7158ce941c258190                          0.0s
 => => extracting sha256:a99f979d704a240405f08913e30302ea667012268e72f3c676dc942e1665f7e4                          0.6s
 => => sha256:8979add51024a9810fb11e34301fcd908b7b83c72d12b40efd49283579f11b19 15.68MB / 15.68MB                  33.1s
 => => extracting sha256:963ffc7a68ff07ad0a48fc7f90d20c556c444bbabfb4bd45a180cdcba6510210                          0.0s
 => => extracting sha256:4d23af0f8da9b6b2cf8447d9208e571426772a573abe3537b5553f9835984251                         94.9s
 => => extracting sha256:8a00ebd09a564c5b434947515b324fe3154f5fd84ef45409580a364ce1accbf1                          1.3s
 => => extracting sha256:f8f1d29d08263d7836d1ac8dcc64734c2198079928f87dc7e35f06b68e9e0e40                          4.4s
 => => extracting sha256:8979add51024a9810fb11e34301fcd908b7b83c72d12b40efd49283579f11b19                          0.4s
 => [books-api-for-e2e base 1/2] FROM mcr.microsoft.com/dotnet/aspnet:8.0@sha256:04f79dd7f0a27eb079c2498ca9bab72  27.9s
 => => resolve mcr.microsoft.com/dotnet/aspnet:8.0@sha256:04f79dd7f0a27eb079c2498ca9bab729ed40b91ed393d24846d6cb5  0.0s
 => => sha256:049861199f76464e65cf70ca24dcc182dcd18808210e0352a21e250199cd5225 1.58kB / 1.58kB                     0.0s
 => => sha256:bb3f2b52e6af242cee1bc6c19ce79e05544f8a1d13f5a6c1e828d98d2dbdc94e 28.06MB / 28.06MB                  21.3s
 => => sha256:11e53e0176cdd626c2032644dddb5a96e6eaf1b0fffe690b7158ce941c258190 3.33kB / 3.33kB                     0.7s
 => => sha256:04f79dd7f0a27eb079c2498ca9bab729ed40b91ed393d24846d6cb505f3fe646 1.08kB / 1.08kB                     0.0s
 => => sha256:3e228f32683d7792cda178c02013d5acf72b58ecb0248fd931b1aca355bb1a94 2.59kB / 2.59kB                     0.0s
 => => sha256:575ae10faf5887a49d34f7131754099bcd5714cb755499cb0a0d3a4de8c9dd2c 18.50MB / 18.50MB                  15.1s
 => => sha256:a99f979d704a240405f08913e30302ea667012268e72f3c676dc942e1665f7e4 30.82MB / 30.82MB                  26.9s
 => => sha256:963ffc7a68ff07ad0a48fc7f90d20c556c444bbabfb4bd45a180cdcba6510210 164B / 164B                        15.3s
 => => sha256:4d23af0f8da9b6b2cf8447d9208e571426772a573abe3537b5553f9835984251 10.74MB / 10.74MB                  17.5s
 => => extracting sha256:bb3f2b52e6af242cee1bc6c19ce79e05544f8a1d13f5a6c1e828d98d2dbdc94e                          1.7s
 => => extracting sha256:575ae10faf5887a49d34f7131754099bcd5714cb755499cb0a0d3a4de8c9dd2c                          0.3s
 => => extracting sha256:a99f979d704a240405f08913e30302ea667012268e72f3c676dc942e1665f7e4                         95.5s
 => => extracting sha256:4d23af0f8da9b6b2cf8447d9208e571426772a573abe3537b5553f9835984251                          0.2s
 => [books-api-for-e2e internal] load build context                                                                0.0s
 => => transferring context: 135.54kB                                                                              0.0s
 => [books-api-for-e2e base 2/2] WORKDIR /app                                                                      0.4s
 => [books-api-for-e2e final 1/2] WORKDIR /app                                                                     0.1s
 => [books-api-for-e2e build 2/7] WORKDIR /src                                                                     1.4s
 => [books-api-for-e2e build 3/7] COPY [Api/Api.csproj, Api/]                                                      0.0s
 => [books-api-for-e2e build 4/7] RUN dotnet restore "Api/Api.csproj"                                             25.2s
 => [books-api-for-e2e build 5/7] COPY . .                                                                         0.1s
 => [books-api-for-e2e build 6/7] WORKDIR /src/Api                                                                 0.1s
 => [books-api-for-e2e build 7/7] RUN dotnet build "Api.csproj" -c Release -o /app/build                          12.4s
 => [books-api-for-e2e publish 1/1] RUN dotnet publish "Api.csproj" -c Release -o /app/publish /p:UseAppHost=fals  5.4s
 => [books-api-for-e2e final 2/2] COPY --from=publish /app/publish .                                               0.4s
 => [books-api-for-e2e] exporting to image                                                                         0.2s
 => => exporting layers                                                                                            0.1s
 => => writing image sha256:b582d7ab4e881486ee3e8c0ee56253cb574fe8d3ae09146beda97a10e71d4064                       0.0s
 => => naming to docker.io/library/inner-circle-books-api-books-api-for-e2e                                        0.0s
 => [books-api-for-e2e] resolving provenance for metadata file                                                     0.0s
[+] Running 4/4
 ✔ Network inner-circle-books-api_books-network  Created                                                           0.0s
 ✔ Container mockServer                          Created                                                           0.1s
 ✔ Container books-db                            Created                                                           0.1s
 ✔ Container books-api-for-e2e                   Created                                                           0.1s
Attaching to books-api-for-e2e, books-db, mockServer
books-db           | The files belonging to this database system will be owned by user "postgres".
books-db           | This user must also own the server process.
books-db           |
books-db           | The database cluster will be initialized with locale "en_US.utf8".
books-db           | The default database encoding has accordingly been set to "UTF8".
books-db           | The default text search configuration will be set to "english".
books-db           |
books-db           | Data page checksums are disabled.
books-db           |
books-db           | fixing permissions on existing directory /var/lib/postgresql/data ... ok
books-db           | creating subdirectories ... ok
books-db           | selecting dynamic shared memory implementation ... posix
books-db           | selecting default max_connections ... 100
books-db           | selecting default shared_buffers ... 128MB
books-db           | selecting default time zone ... Etc/UTC
books-db           | creating configuration files ... ok
books-db           | running bootstrap script ... ok
books-db           | performing post-bootstrap initialization ... ok
books-api-for-e2e  | Unhandled exception. Npgsql.NpgsqlException (0x80004005): Failed to connect to 172.18.0.3:5432
books-api-for-e2e  |  ---> System.Net.Sockets.SocketException (111): Connection refused
books-api-for-e2e  |    at System.Net.Sockets.Socket.AwaitableSocketAsyncEventArgs.ThrowException(SocketError error, CancellationToken cancellationToken)
books-api-for-e2e  |    at System.Net.Sockets.Socket.AwaitableSocketAsyncEventArgs.System.Threading.Tasks.Sources.IValueTaskSource.GetResult(Int16 token)
books-api-for-e2e  |    at System.Net.Sockets.Socket.<ConnectAsync>g__WaitForConnectWithCancellation|285_0(AwaitableSocketAsyncEventArgs saea, ValueTask connectTask, CancellationToken cancellationToken)
books-api-for-e2e  |    at Npgsql.TaskTimeoutAndCancellation.ExecuteAsync(Func`2 getTaskFunc, NpgsqlTimeout timeout, CancellationToken cancellationToken)
books-api-for-e2e  |    at Npgsql.Internal.NpgsqlConnector.ConnectAsync(NpgsqlTimeout timeout, CancellationToken cancellationToken)
books-api-for-e2e  |    at Npgsql.Internal.NpgsqlConnector.ConnectAsync(NpgsqlTimeout timeout, CancellationToken cancellationToken)
books-api-for-e2e  |    at Npgsql.Internal.NpgsqlConnector.RawOpen(SslMode sslMode, NpgsqlTimeout timeout, Boolean async, CancellationToken cancellationToken, Boolean isFirstAttempt)
books-api-for-e2e  |    at Npgsql.Internal.NpgsqlConnector.<Open>g__OpenCore|213_1(NpgsqlConnector conn, SslMode sslMode, NpgsqlTimeout timeout, Boolean async, CancellationToken cancellationToken, Boolean isFirstAttempt)
books-api-for-e2e  |    at Npgsql.Internal.NpgsqlConnector.Open(NpgsqlTimeout timeout, Boolean async, CancellationToken cancellationToken)
books-api-for-e2e  |    at Npgsql.UnpooledDataSource.Get(NpgsqlConnection conn, NpgsqlTimeout timeout, Boolean async, CancellationToken cancellationToken)
books-api-for-e2e  |    at Npgsql.NpgsqlConnection.<Open>g__OpenAsync|42_0(Boolean async, CancellationToken cancellationToken)
books-api-for-e2e  |    at Microsoft.EntityFrameworkCore.Storage.RelationalConnection.OpenInternalAsync(Boolean errorsExpected, CancellationToken cancellationToken)
books-api-for-e2e  |    at Microsoft.EntityFrameworkCore.Storage.RelationalConnection.OpenInternalAsync(Boolean errorsExpected, CancellationToken cancellationToken)
books-api-for-e2e  |    at Microsoft.EntityFrameworkCore.Storage.RelationalConnection.OpenAsync(CancellationToken cancellationToken, Boolean errorsExpected)
books-api-for-e2e  |    at Npgsql.EntityFrameworkCore.PostgreSQL.Storage.Internal.NpgsqlDatabaseCreator.Exists(Boolean async, CancellationToken cancellationToken)
books-api-for-e2e  |    at Npgsql.EntityFrameworkCore.PostgreSQL.Storage.Internal.NpgsqlDatabaseCreator.Exists(Boolean async, CancellationToken cancellationToken)
books-api-for-e2e  |    at Microsoft.EntityFrameworkCore.Migrations.HistoryRepository.ExistsAsync(CancellationToken cancellationToken)
books-api-for-e2e  |    at Microsoft.EntityFrameworkCore.Migrations.HistoryRepository.GetAppliedMigrationsAsync(CancellationToken cancellationToken)
books-api-for-e2e  |    at Npgsql.EntityFrameworkCore.PostgreSQL.Migrations.Internal.NpgsqlMigrator.MigrateAsync(String targetMigration, CancellationToken cancellationToken)
books-api-for-e2e  |    at Program.<Main>$(String[] args) in /src/Api/Program.cs:line 96
books-api-for-e2e  |    at Program.<Main>(String[] args)
books-db           | initdb: warning: enabling "trust" authentication for local connections
books-db           | You can change this by editing pg_hba.conf or using the option -A, or
books-db           | --auth-local and --auth-host, the next time you run initdb.
books-db           | syncing data to disk ... ok
books-db           |
books-db           |
books-db           | Success. You can now start the database server using:
books-db           |
books-db           |     pg_ctl -D /var/lib/postgresql/data -l logfile start
books-db           |
books-api-for-e2e exited with code 133
books-db           | waiting for server to start....2024-12-18 04:47:05.137 UTC [48] LOG:  starting PostgreSQL 14.13 (Debian 14.13-1.pgdg120+1) on aarch64-unknown-linux-gnu, compiled by gcc (Debian 12.2.0-14) 12.2.0, 64-bit
books-db           | 2024-12-18 04:47:05.140 UTC [48] LOG:  listening on Unix socket "/var/run/postgresql/.s.PGSQL.5432"
books-db           | 2024-12-18 04:47:05.156 UTC [49] LOG:  database system was shut down at 2024-12-18 04:47:03 UTC
books-db           | 2024-12-18 04:47:05.169 UTC [48] LOG:  database system is ready to accept connections
books-db           |  done
books-db           | server started
books-db           | CREATE DATABASE
books-db           |
books-db           |
books-db           | /usr/local/bin/docker-entrypoint.sh: ignoring /docker-entrypoint-initdb.d/*
books-db           |
books-db           | waiting for server to shut down....2024-12-18 04:47:05.804 UTC [48] LOG:  received fast shutdown request
books-db           | 2024-12-18 04:47:05.805 UTC [48] LOG:  aborting any active transactions
books-db           | 2024-12-18 04:47:05.807 UTC [48] LOG:  background worker "logical replication launcher" (PID 55) exited with exit code 1
books-db           | 2024-12-18 04:47:05.808 UTC [50] LOG:  shutting down
books-db           | 2024-12-18 04:47:05.842 UTC [48] LOG:  database system is shut down
books-db           |  done
books-db           | server stopped
books-db           |
books-db           | PostgreSQL init process complete; ready for start up.
books-db           |
books-db           | 2024-12-18 04:47:05.935 UTC [1] LOG:  starting PostgreSQL 14.13 (Debian 14.13-1.pgdg120+1) on aarch64-unknown-linux-gnu, compiled by gcc (Debian 12.2.0-14) 12.2.0, 64-bit
books-db           | 2024-12-18 04:47:05.935 UTC [1] LOG:  listening on IPv4 address "0.0.0.0", port 5432
books-db           | 2024-12-18 04:47:05.935 UTC [1] LOG:  listening on IPv6 address "::", port 5432
books-db           | 2024-12-18 04:47:05.940 UTC [1] LOG:  listening on Unix socket "/var/run/postgresql/.s.PGSQL.5432"
books-db           | 2024-12-18 04:47:05.965 UTC [63] LOG:  database system was shut down at 2024-12-18 04:47:05 UTC
books-db           | 2024-12-18 04:47:05.971 UTC [1] LOG:  database system is ready to accept connections
mockServer         | 2024-12-18 04:47:06 5.15.0 INFO using environment variables:
mockServer         |
mockServer         |   [
mockServer         |    SERVER_PORT=1080,
mockServer         |    MOCKSERVER_INITIALIZATION_JSON_PATH=/config/initializer.json
mockServer         |   ]
mockServer         |
mockServer         |  and system properties:
mockServer         |
mockServer         |   [
mockServer         |    mockserver.propertyFile=/config/mockserver.properties
mockServer         |   ]
mockServer         |
mockServer         |  and command line options:
mockServer         |
mockServer         |   [
mockServer         |    logLevel=INFO
mockServer         |   ]
mockServer         |
mockServer         | 2024-12-18 04:47:07 5.15.0 INFO loading JSON initialization file:
mockServer         |
mockServer         |   /config/initializer.json
mockServer         |`

**Review by akovylyaeva — APPROVED** [review]
